src/main/java/uk/nhs/tis/trainee/notifications/service/NotificationService.java

```diff
@@ -354,4 +359,17 @@ public boolean programmeMembershipIsNotifiable(ProgrammeMembership programmeMemb
     String traineeId = programmeMembership.getPersonId();
     return messagingControllerService.isValidRecipient(traineeId, messageType);
   }
+
+  /**
+   * Validate the stored GMC number of the trainee.
+   *
+   * @param gmcNumber The GMC number to validate.
+   * @return true if it is 7 consecutive numerical digits string, otherwise false.
+   */
+  public boolean isValidGmc(String gmcNumber) {
+    if (gmcNumber == null) {
+      return false;
+    }
+    return (gmcNumber.length() == 7 && StringUtils.isNumeric(gmcNumber));
```

Comment:
I think you need to trim the gmcNumber, there are a fair number with a trailing space.
Also, I think James' rule was 'must contain 7 consecutive digits', not that it could only be 7 digits, afaik. I can't find the comment thread unfortunately to double-check, but maybe ping James to confirm?

Reply:
As confirmed by James on the Slack thread, a GMC number should be no more nor no less than 7 numerical digits. I will leave the logic as is then.